feature/settings/src/main/java/com/google/jetpackcamera/settings/SettingEnabledState.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.jetpackcamera.settings
+
+import com.google.jetpackcamera.settings.ui.DEVICE_UNSUPPORTED_TAG
+import com.google.jetpackcamera.settings.ui.FPS_UNSUPPORTED_TAG
+import com.google.jetpackcamera.settings.ui.LENS_UNSUPPORTED_TAG
+import com.google.jetpackcamera.settings.ui.STABILIZATION_UNSUPPORTED_TAG
+
+val deviceUnsupported =
+    SettingEnabledState.Disabled(disabledRationale = setOf(DisabledRationale.DEVICE_UNSUPPORTED))
+
+sealed interface SettingUiState {
+    val settingEnabledState: SettingEnabledState
+}
+
+sealed interface SettingEnabledState {
+    val isEnabled: Boolean
+
+    data class Disabled(
+        val disabledRationale: Set<DisabledRationale>
+    ) : SettingEnabledState {
+        override val isEnabled: Boolean = false
+    }
+
+    data object Enabled : SettingEnabledState {
+        override val isEnabled: Boolean = true
+    }
+}
+
+class FpsUiState(
+    override val settingEnabledState: SettingEnabledState,
+    val optionsStates: Map<Int, SettingEnabledState>
+) : SettingUiState
+
+class FlipLensUiState(override val settingEnabledState: SettingEnabledState) : SettingUiState
+class StabilizationUiState(
+    override val settingEnabledState: SettingEnabledState,
+    val previewStabilizationState: SettingEnabledState,
+    val videoStabilizationState: SettingEnabledState
+) : SettingUiState
```

Comment:
One thing we want to avoid in a sealed class heirarchy is properties that are not used when we are in a certain state. So in this case, it doesn't make sense to have `isEnabled` as a property when we already have sealed interface subclasses for `Enabled` and `Disabled`. Additionally, in `FpsUiState`, we have `optionsStates` which may not be necessary when `settingsEnabledState` is `Disabled`.

Consider a hierarchy like this:
```
sealed interface SettingUiState {
    data object Disabled : SettingsUiState
    data class Enabled(
        val defaultFrontLensUiState: DefaultFrontLensUiState,
        val fpsUiState: FpsUiState,
        val stabilizationUiState: StabilizationUiState
       // ... Additional properties for each setting
    ) : SettingUiState
}

sealed interface SingleSelectableState {
    data object Selectable : SingleSelectableState
    data class Disabled(val disabledRationale: DisabledRationale) : SingleSelectableState
}

// Doesn't really need to be a sealed interface since even when disabled,
// we still need the radio button to either be on or off.
data class DefaultLensFacingUiState(
    val useFrontLens: Boolean,
    val selectable: SingleSelectableState
)

sealed interface FpsUiState {
    data class Enabled(
        val currentSelection: Int,
        val fpsAutoState: SingleSelectableState,
        val fpsFifteenState: SingleSelectableState,
        val fpsThirtyState: SingleSelectableState,
        val fpsSixtyState: SingleSelectableState,
        // Contains text like "Selected FPS only supported by rear lens"
        val additionalContext: String = "",
    ) : FpsUiState

    // FPS selection completely disabled. Cannot open dialog.
    data class Disabled(val disabledRationale: DisabledRationale) : FpsUiState
}

sealed interface StabilizationUiState {
    data class Enabled(
        val currentSelection: Int,
        val stabilizationOnState: SingleSelectableState,
        val stabilizationHighQualityState: SingleSelectableState,
        // Contains text like "Selected stabilization mode only supported by rear lens"
        val additionalContext: String = ""
    ) : StabilizationUiState

    // Stabilization selection completely disabled. Cannot open dialog.
    data class Disabled(val disabledRationale: DisabledRationale) : StabilizationUiState
}

data class DisabledRationale(val testTag: String, val reasonText: String)
```

If we really wanted to, we could reduce duplication by turning `FpsUiState` and `StabilizationState` into a shared generic sealed interface such as `RadioButtonSettingUiState`, but then we would also need to replace all of the properties with something like a `Map`. As it is, it's a bit more readable and is more flexible if we need to add special behavior to the individual settings.